relayer/relays/beacon/header/syncer/syncer.go

```diff
@@ -525,53 +543,60 @@ func (s *Syncer) GetHeaderUpdateBySlotWithCheckpoint(slot uint64, checkpoint *ca
 
 func (s *Syncer) GetHeaderUpdate(blockRoot common.Hash, checkpoint *cache.Proof) (scale.HeaderUpdatePayload, error) {
 	var update scale.HeaderUpdatePayload
-	blockResponse, err := s.Client.GetBeaconBlock(blockRoot)
+	blockBytes, err := s.Client.GetBeaconBlockBytes(blockRoot)
 	if err != nil {
 		return update, fmt.Errorf("fetch block: %w", err)
 	}
-	data := blockResponse.Data.Message
-	slot, err := util.ToUint64(data.Slot)
+
+	header, err := s.Client.GetHeaderByBlockRoot(blockRoot)
 	if err != nil {
-		return update, err
+		return scale.HeaderUpdatePayload{}, fmt.Errorf("fetch block: %w", err)
 	}
 
-	sszBlock, err := blockResponse.ToFastSSZ(s.protocol.DenebForked(slot))
-	if err != nil {
-		return update, err
+	slot := header.Slot
+
+	var signedBlock state.SignedBeaconBlock
+	forkVersion := s.protocol.ForkVersion(slot)
+	if forkVersion == protocol.Electra {
+		signedBlock = &state.SignedBeaconBlockElectra{}
+	} else {
+		signedBlock = &state.SignedBeaconBlockDeneb{}
 	}
 
-	header, err := s.Client.GetHeaderBySlot(sszBlock.GetBeaconSlot())
+	err = signedBlock.UnmarshalSSZ(blockBytes)
 	if err != nil {
-		return scale.HeaderUpdatePayload{}, fmt.Errorf("fetch block: %w", err)
+		return scale.HeaderUpdatePayload{}, fmt.Errorf("unmarshal block ssz: %w", err)
 	}
 
 	beaconHeader, err := header.ToScale()
 	if err != nil {
 		return scale.HeaderUpdatePayload{}, fmt.Errorf("beacon header to scale: %w", err)
 	}
 
-	executionHeaderBranch, err := s.getExecutionHeaderBranch(sszBlock)
+	beaconBlock := signedBlock.GetBlock()
+	executionHeaderBranch, err := s.getExecutionHeaderBranch(beaconBlock)
 	if err != nil {
 		return scale.HeaderUpdatePayload{}, err
 	}
 
 	var versionedExecutionPayloadHeader scale.VersionedExecutionPayloadHeader
-	if s.protocol.DenebForked(slot) {
-		executionPayloadScale, err := api.DenebExecutionPayloadToScale(sszBlock.ExecutionPayloadDeneb())
+	// The execution payload did not change in Electra, so we reuse Deneb's version.
+	if forkVersion == protocol.Electra || forkVersion == protocol.Deneb {
+		executionPayloadScale, err := api.DenebExecutionPayloadToScale(beaconBlock.ExecutionPayloadDeneb())
 		if err != nil {
 			return scale.HeaderUpdatePayload{}, err
 		}
 		versionedExecutionPayloadHeader = scale.VersionedExecutionPayloadHeader{Deneb: &executionPayloadScale}
 	} else {
-		executionPayloadScale, err := api.CapellaExecutionPayloadToScale(sszBlock.ExecutionPayloadCapella())
+		executionPayloadScale, err := api.CapellaExecutionPayloadToScale(beaconBlock.ExecutionPayloadCapella())
```

Comment:
Can just remove the Capella branch?

Reply:
Good idea, https://github.com/Snowfork/snowbridge/pull/1283/commits/ac5203550bd1df76cfc569c88c77cad6cbae593b.